### PR TITLE
LPS-139129 Add permission checks to remote apps

### PR DESCRIPTION
**Pull request by javierdearcos**
This PR adds the permissions to the RemoteAppEntry entity and RemoteAppAdmin portlet.

The first thing to review is if the defined permissions make sense in the context of the Remote Apps application and use cases. Check [first commit](https://github.com/liferay-frontend/liferay-portal/commit/31f16a16d0a48d5e1989f80813178c62065a8823)

Taking other entities as an example I defined the ADD_ENTRY and PERMISSIONS actions for the collection. Checking the frontend code I didn't see any place where the permissions could be changed, but I add the PERMISSIONS action in case it is interesting to add it in the future. If not, we can remove it.

From the entity level, the actions defined are DELETE, PERMISSIONS, UPDATE, and VIEW. The VIEW permission is given by a guest role.

I've modified the Service impl to check permissions before calling the local service to execute the desired action and the MVCAction commands to always use the service (and avoid using the local service which will avoid the permission checking).

I'm not adding any label to this PR although I think that s-dxp is the best fit. Please add it when you review the PR.

### CODE REVIEW

**Review by nhpatt — APPROVED** [review]
LGTM :D

👏 👏 👏

**Comment by nhpatt** [comment]
Sent directly to Brian here: https://github.com/brianchandotcom/liferay-portal/pull/107480

I'll add more feedback later but nice work :)

File: modules/apps/remote-app/remote-app-api/src/main/java/com/liferay/remote/app/service/persistence/RemoteAppEntryUtil.java [hunk 1]
```diff
@@ -452,6 +529,79 @@ public static RemoteAppEntry fetchByUuid_C_Last(
...
+	 * Returns the remote app entries before and after the current remote app entry in the ordered set of remote app entries that the user has permission to view where uuid = &#63; and companyId = &#63;.
+	 *
+	 * @param remoteAppEntryId the primary key of the current remote app entry
+	 * @param uuid the uuid
+	 * @param companyId the company ID
+	 * @param orderByComparator the comparator to order the set by (optionally <code>null</code>)
+	 * @return the previous, current, and next remote app entry
+	 * @throws NoSuchRemoteAppEntryException if a remote app entry with the primary key could not be found
+	 */
+	public static RemoteAppEntry[] filterFindByUuid_C_PrevAndNext(
```
Comment (diegonvs) [inline modules/apps/remote-app/remote-app-api/src/main/java/com/liferay/remote/app/service/persistence/RemoteAppEntryUtil.java:596]: Hey Javi, 

What means this `_C` in several signatures?

Comment (nhpatt) [reply modules/apps/remote-app/remote-app-api/src/main/java/com/liferay/remote/app/service/persistence/RemoteAppEntryUtil.java:596]: `Company`, that is a generated method (by service builder) and the naming abbreviates each param.

**Comment by nhpatt** [comment]
Ups, I did not close it although it was already sent to Brian and merged, closing...

**Pull request closed without merging**